src/ray/core_worker/common.h

```diff
@@ -66,6 +68,45 @@ class TaskArg {
   const std::shared_ptr<Buffer> data_;
 };
 
+/// Task specification, which includes the immutable information about the task
+/// which are determined at the submission time.
+class TaskSpec {
```

Comment:
This has some overlap with the `Task` in `task.h`. Can we just use that one?

Reply:
Currently `Task` is only used by raylet, but it's not used by worker. `Task` has `TaskSepc` and `TaskExecutionSpec`, the latter currently has some information that are updated dynamically (number of forwards, timestamp), and I'm feeling it's not quite appropriate to include these when submitting a task. This could be addressed by a later PR.

Reply:
ok, we can remove this once we move everything to protobuf